drivers/sensor/bosch/bmp581/bmp581_decoder.c

```diff
@@ -105,53 +110,40 @@ static int bmp581_decoder_get_size_info(struct sensor_chan_spec chan_spec,
 	}
 }
 
-static int bmp581_decoder_decode(const uint8_t *buffer,
-				struct sensor_chan_spec chan_spec,
-				uint32_t *fit,
-				uint16_t max_count,
-				void *data_out)
+static int bmp581_convert_raw_to_q31_value(const struct bmp581_encoded_header *header,
+					   struct sensor_chan_spec *chan_spec,
+					   const struct bmp581_frame *frame,
+					   uint32_t *fit,
+					   struct sensor_q31_data *out)
 {
-	const struct bmp581_encoded_data *edata = (const struct bmp581_encoded_data *)buffer;
-	uint8_t channel_request;
-
-	if (*fit != 0) {
-		return 0;
-	}
-
-	if (max_count == 0 || chan_spec.chan_idx != 0) {
-		return -EINVAL;
-	}
-
-	channel_request = bmp581_encode_channel(chan_spec.chan_type);
-	if ((channel_request & edata->header.channels) != channel_request) {
+	if (((header->events & BMP581_EVENT_FIFO_WM) != 0) && (*fit >= header->fifo_count)) {
 		return -ENODATA;
+	} else if (((header->events & BMP581_EVENT_FIFO_WM) == 0) && (*fit != 0)) {
+		return -ENODATA;
```

Comment:
the two conditions are not really related right? just make two separate ifs

Reply:
or, if the two conditions are related, they have the same outcome, make it a single condition

Reply:
Addressed.

Reply:
I'm not sure I'd classify this change as "way better" at some point these boolean branch expressions spanning multiple lines become unreadable.